src/util/formDevices.tsx

```diff
@@ -232,3 +233,36 @@ export const deduplicateName = (
   }
   return candidate;
 };
+
+export const addNicDevice = ({
+  formik,
+  deviceName,
+  deviceNetworkName,
+}: {
+  formik: InstanceAndProfileFormikProps;
+  deviceName: string;
+  deviceNetworkName: string;
+}) => {
+  const copy = [...formik.values.devices];
+  copy.push({
+    type: "nic",
+    name: deviceName,
+    network: deviceNetworkName,
+  });
+  formik.setFieldValue("devices", copy);
+};
+
+export const focusNetwork = (id: number) => {
```

Comment:
```suggestion
export const focusNicDevice = (id: number) => {
```
A bit more consistent naming.